ledger/acctupdates.go

```diff
@@ -2138,28 +2166,51 @@ func (au *accountUpdates) commitRound(offset uint64, dbRound basics.Round, lookb
 
 		db.ResetTransactionWarnDeadline(ctx, tx, time.Now().Add(accountsUpdatePerRoundHighWatermark*time.Duration(offset)))
 
+		if updateStats {
+			stats.OldAccountPreloadDuration = time.Duration(time.Now().UnixNano())
+		}
+
 		err = compactDeltas.accountsLoadOld(tx)
 		if err != nil {
 			return err
 		}
 
+		if updateStats {
+			stats.OldAccountPreloadDuration = time.Duration(time.Now().UnixNano()) - stats.OldAccountPreloadDuration
+		}
+
 		err = totalsNewRounds(tx, deltas[:offset], compactDeltas, roundTotals[1:offset+1], config.Consensus[consensusVersion])
 		if err != nil {
 			return err
 		}
 
+		if updateStats {
+			stats.MerkleTrieUpdateDuration = time.Duration(time.Now().UnixNano())
+		}
+
 		err = au.accountsUpdateBalances(compactDeltas)
 		if err != nil {
 			return err
 		}
 
+		if updateStats {
```

Comment:
when having back-to-back timing, use the *same* value you received from `time.Now()`.
This would allow you to ensure there is no time-slippage between the different blocks, and that they always sum up.

Reply:
do you mean use time.Now().UnixNano() instead of time.Duration(time.Now().UnixNano())?

Reply:
do you mean use time.Now().UnixNano() instead of time.Duration(time.Now().UnixNano())?

Reply:
no, I mean -
```golang
if updateStats {
     now := time.Duration(time.Now().UnixNano())
     stats.MerkleTrieUpdateDuration = now - stats.MerkleTrieUpdateDuration
     stats.AccountsWritingDuration = now
}
```
so that it would be the *same* `time.Now()`